include/nbl/builtin/hlsl/spirv_intrinsics/core.hlsl

```diff
@@ -47,199 +69,1272 @@ static const uint32_t3 LocalInvocationId;
+T groupNonUniformLogicalOr_GroupNonUniformClustered(uint32_t executionScope, [[vk::ext_literal]] uint32_t operation, T value);
+
+template<typename T>
+[[vk::ext_capability(spv::CapabilityGroupNonUniformPartitionedNV)]]
+[[vk::ext_instruction(spv::OpGroupNonUniformLogicalOr)]]
+T groupNonUniformLogicalOr_GroupNonUniformPartitionedNV(uint32_t executionScope, [[vk::ext_literal]] uint32_t operation, T value);
+
+template<typename T>
+[[vk::ext_capability(spv::CapabilityGroupNonUniformArithmetic)]]
+[[vk::ext_instruction(spv::OpGroupNonUniformLogicalXor)]]
+T groupNonUniformLogicalXor_GroupNonUniformArithmetic(uint32_t executionScope, [[vk::ext_literal]] uint32_t operation, T value);
+
+template<typename T>
+[[vk::ext_capability(spv::CapabilityGroupNonUniformClustered)]]
+[[vk::ext_instruction(spv::OpGroupNonUniformLogicalXor)]]
+T groupNonUniformLogicalXor_GroupNonUniformClustered(uint32_t executionScope, [[vk::ext_literal]] uint32_t operation, T value);
+
+template<typename T>
+[[vk::ext_capability(spv::CapabilityGroupNonUniformPartitionedNV)]]
+[[vk::ext_instruction(spv::OpGroupNonUniformLogicalXor)]]
+T groupNonUniformLogicalXor_GroupNonUniformPartitionedNV(uint32_t executionScope, [[vk::ext_literal]] uint32_t operation, T value);
+
+template<typename T>
+[[vk::ext_capability(spv::CapabilityGroupNonUniformQuad)]]
+[[vk::ext_instruction(spv::OpGroupNonUniformQuadBroadcast)]]
+T groupNonUniformQuadBroadcast(uint32_t executionScope, T value, uint32_t index);
+
+template<typename T>
+[[vk::ext_capability(spv::CapabilityQuadControlKHR)]]
+[[vk::ext_instruction(spv::OpGroupNonUniformQuadAllKHR)]]
+T groupNonUniformQuadAllKHR(bool predicate);
+
+template<typename T>
+[[vk::ext_capability(spv::CapabilityQuadControlKHR)]]
+[[vk::ext_instruction(spv::OpGroupNonUniformQuadAnyKHR)]]
+T groupNonUniformQuadAnyKHR(bool predicate);
 
-template<typename Unsigned>
-[[vk::ext_instruction( spv::OpBitFieldUExtract )]]
-enable_if_t<is_unsigned_v<Unsigned>, Unsigned> bitFieldUExtract( Unsigned val, uint32_t offsetBits, uint32_t numBits );
+template<typename T>
+[[vk::ext_capability(spv::CapabilityGroupNonUniformPartitionedNV)]]
+[[vk::ext_instruction(spv::OpGroupNonUniformPartitionNV)]]
+T groupNonUniformPartitionNV(T value);
+
+[[vk::ext_capability(spv::CapabilityAtomicFloat16MinMaxEXT)]]
+[[vk::ext_instruction(spv::OpAtomicFMinEXT)]]
+float atomicMinEXT_AtomicFloat16MinMaxEXT([[vk::ext_reference]] float pointer, uint32_t memoryScope,  uint32_t semantics, float value);
+
+[[vk::ext_capability(spv::CapabilityAtomicFloat32MinMaxEXT)]]
+[[vk::ext_instruction(spv::OpAtomicFMinEXT)]]
+float atomicMinEXT_AtomicFloat32MinMaxEXT([[vk::ext_reference]] float pointer, uint32_t memoryScope,  uint32_t semantics, float value);
+
+[[vk::ext_capability(spv::CapabilityAtomicFloat64MinMaxEXT)]]
+[[vk::ext_instruction(spv::OpAtomicFMinEXT)]]
+float atomicMinEXT_AtomicFloat64MinMaxEXT([[vk::ext_reference]] float pointer, uint32_t memoryScope,  uint32_t semantics, float value);
```

Comment:
wrong types, first should be `float16_t` second should be `float32_t`, last should be `float64_t`

you should also have overloads for `template<typename P>` for a spir-v pointer